src/databricks/labs/ucx/hive_metastore/tables.py

```diff
@@ -481,3 +481,134 @@ def _describe(self, catalog: str, database: str, table: str) -> Table | None:
         except Exception as e:  # pylint: disable=broad-exception-caught
             logger.error(f"Couldn't fetch information for table {full_name} : {e}")
             return None
+
+
+class FasterTableScanCrawler(CrawlerBase):
+    def __init__(self, backend: SqlBackend, schema, include_databases: list[str] | None = None):
+        self._backend = backend
+        self._include_database = include_databases
+
+        # pylint: disable-next=import-error,import-outside-toplevel
+        from pyspark.sql.session import SparkSession  # type: ignore[import-not-found]
+
+        super().__init__(backend, "hive_metastore", schema, "tables", Table)
+        self._spark = SparkSession.builder.getOrCreate()
+
+    def snapshot(self) -> list[Table]:
+        """
+        Takes a snapshot of tables in the specified catalog and database.
+
+        Returns:
+            list[Table]: A list of Table objects representing the snapshot of tables.
+        """
+        return self._snapshot(partial(self._try_load), partial(self._crawl))
+
+    def _try_load(self) -> Iterable[Table]:
+        """Tries to load table information from the database or throws TABLE_OR_VIEW_NOT_FOUND error"""
+        for row in self._fetch(f"SELECT * FROM {escape_sql_identifier(self.full_name)}"):
+            yield Table(*row)
+
+    @cached_property
+    def _external_catalog(self):
+        return self._spark._jsparkSession.sharedState().externalCatalog()
+
+    def _iterator(self, result: typing.Any) -> Iterator:
+        it = result.iterator()
+        while it.hasNext():
+            yield it.next()
+
+    def _all_databases(self) -> list[str]:
+        if not self._include_database:
+            return list(self._iterator(self._external_catalog.listDatabases()))
+        return self._include_database
+
+    def _list_tables(self, database: str) -> list[str] | None:
+        try:
+            return list(self._iterator(self._external_catalog.listTables(database)))
+        except Exception as err:
+            logger.warning(f"Failed to list tables in {database}: {err}")
+            return None
```

Comment:
```suggestion
    def _list_tables(self, database: str) -> list[str]:
        try:
            return list(self._iterator(self._external_catalog.listTables(database)))
        except Exception as err:
            logger.warning(f"Failed to list tables in {database}: {err}")
            return []
```
avoid `| None` whenever you can.